server/src/controllers/asset-media.controller.ts

```diff
@@ -42,10 +56,59 @@ export class AssetMediaController {
     private service: AssetMediaService,
   ) {}
 
```

Comment:
use getFiles() and call it `createAsset` instead of `uploadFile`
```suggestion
@Post()
  @UseInterceptors(AssetMediaUploadInterceptor, FileUploadInterceptor)
  @ApiConsumes('multipart/form-data')
  @ApiHeader({
    name: ImmichHeader.CHECKSUM,
    description: 'sha1 checksum that can be used for duplicate detection before the file is uploaded',
    required: false,
  })
  @ApiBody({
    type: CreateAssetMediaDto,
  })
  @Authenticated({ sharedLink: true })
  @EndpointLifecycle({ addedAt: 'v1.106.0' })
  async createAsset(
    @Auth() auth: AuthDto,
    @UploadedFiles(new ParseFilePipe({ validators: [new FileNotEmptyValidator(['assetData'])] })) files: UploadFiles,
    @Body() dto: AssetMediaReplaceDto,
    @Res({ passthrough: true }) res: Response,
  ): Promise<AssetMediaResponseDto> {
    const { file, sidecarFile } = getFiles(files);
    const assetMediaResponse = await this.service.createAsset(auth, dto, file, sidecarFile);
    if (assetMediaResponse.status === AssetMediaStatusEnum.DUPLICATE) {
      res.status(HttpStatus.OK);
    }
    return assetMediaResponse;
  }
```

Reply:
I have made most of these changes. I might make the "remove live photo data" in a follow-up PR though.

Reply:
Ended up doing it all in this PR.